contracts/extensions/OneTxPayment.sol

```diff
@@ -98,44 +145,92 @@ contract OneTxPayment {
   /// @param _childSkillIndex Index of the _permissionDomainId skill.children array to get
   /// @param _callerPermissionDomainId The domainId in which the _caller_ has permissions to add a payment and fund it
   /// @param _callerChildSkillIndex Index of the _callerPermissionDomainId skill.children array to get
-  /// @param _worker The address of the recipient of the payment
-  /// @param _token The address of the token the payment is being made in. 0x00 for Ether.
-  /// @param _amount The amount of the token being paid out
+  /// @param _workers The addresses of the recipients of the payment
+  /// @param _tokens The addresses of the token the payments are being made in. 0x00 for Ether.
+  /// @param _amounts The amounts of the tokens being paid out
   /// @param _domainId The Id of the domain the payment should be coming from
   /// @param _skillId The Id of the skill that the payment should be marked with, possibly awarding reputation in this skill.
   function makePaymentFundedFromDomain(
     uint256 _permissionDomainId,
     uint256 _childSkillIndex,
     uint256 _callerPermissionDomainId,
     uint256 _callerChildSkillIndex,
-    address payable _worker,
-    address _token,
-    uint256 _amount,
+    address payable[] memory  _workers,
+    address[] memory _tokens,
+    uint256[] memory _amounts,
     uint256 _domainId,
     uint256 _skillId) public
   {
     // Check caller is able to call {add,finalize}Payment and moveFundsBetweenPots on the colony
     validateCallerPermissions(_callerPermissionDomainId, _callerChildSkillIndex, _domainId);
-
-    // Add a new payment
-    uint256 paymentId = colony.addPayment(_permissionDomainId, _childSkillIndex, _worker, _token, _amount, _domainId, _skillId);
-    ColonyDataTypes.Payment memory payment = colony.getPayment(paymentId);
-    ColonyDataTypes.Domain memory domain = colony.getDomain(_domainId);
-
-    // Fund the payment
-    colony.moveFundsBetweenPots(
-      _permissionDomainId,
-      _childSkillIndex,
-      _childSkillIndex,
-      domain.fundingPotId,
-      payment.fundingPotId,
-      _amount,
-      _token
+    //Arrays must be of equal size
+    require(
+      _workers.length == _tokens.length && _workers.length == _amounts.length,
+      "funding-arrays-must-be-equal-sizes"
     );
-    colony.finalizePayment(_permissionDomainId, _childSkillIndex, paymentId);
-
-    // Claim payout on behalf of the recipient
-    colony.claimPayment(paymentId, _token);
+    //Get around stack too deep with array
+    //index[0] = expenditure slot
+    //index[1] = expenditureId/paymentId
+    //index[2] = fundingPotId
```

Comment:
Not sure which one you mean. 

`index[0]:`
 -  The slot variable, is just a looping variable to retain which slot is the current slot. 

`index[1]:`
 -  This stores IDs so I only need one implementation of `moveFundsBetweenPots()`. Otherwise I'd need two variable storing the IDs and two `moveFundsBetweenPots()`

`index[2]:`
 -  stores `fundingPotId`  for similar reasons to index[1]. Saves variables and no reused code.